homeassistant/components/media_player/samsungtv.py

```diff
@@ -122,8 +129,28 @@ def __init__(self, host, port, name, timeout, mac):
 
     def update(self):
         """Retrieve the latest data."""
-        # Send an empty key to see if we are still connected
-        self.send_key('KEY')
+        if self._config['method'] == 'websocket' and self._status_via_rest:
+            # Check if the TV Rest Server is active
+            self._update_status_via_rest()
+        else:
+            # Send an empty key to see if we are still connected
+            self.send_key('KEY')
+
+    def _update_status_via_rest(self):
+        url = "http://{}:{}/api/v2/"
```

Comment:
This logic needs to be added to the upstream `samsungctl` library.

Reply:
I'm afraid that would be out of scope for this library, since the library just focus on control via websocket, and does not interact with the REST endpoint for now.

If it's required I'll try, but not sure my code make it to `samsungctl` library.

Reply:
Pull request opened in SamsungCTL library
https://github.com/Ape/samsungctl/pull/71